lib/Conversion/DCToHW/DCToHW.cpp

```diff
@@ -255,7 +255,7 @@ struct RTLBuilder {
            "signal must be provided to the reg(...) function.");
 
     return b.create<seq::CompRegOp>(loc, in.getType(), in, resolvedClk, name,
-                                    resolvedRst, rstValue, StringAttr());
+                                    resolvedRst, rstValue, hw::InnerSymAttr());
```

Comment:
Trailing optional not needed?

Reply:
Looks like optional arguments are still needed, it just adds a `/*optional*/` comment to the builder argument and allows you to pass in null :/

Reply:
I wish MLIR had something like the [builder pattern](https://refactoring.guru/design-patterns/builder/rust/example) to construct ops. Some of them have quite a large number of optional operands or attributes where that pattern would provide easier to read code.